### PR TITLE
Properly handle links while extracting files 

### PR DESCRIPTION
**Pull request by azubieta**
Add support to extract hard links on `appimage_extract_file_following_symlinks`

### CODE REVIEW

File: src/libappimage/libappimage.c [hunk 1]
```diff
@@ -1890,12 +1888,36 @@ void appimage_create_thumbnail(const char *appimage_file_path, bool verbose) {
...
+
+        if (data.link) {
+            if (data.link && visited_entries &&
+                g_slist_find_custom(visited_entries, data.link, (GCompareFunc) strcmp))
+                looping = true;
+
+            g_remove(target_file_path);
+        }
+
+    } while (data.link && !looping);
```
Comment (TheAssassin) [inline src/libappimage/libappimage.c]: Is `data.link` a boolean? If not, please use a more explicit syntax.

Comment (azubieta) [reply src/libappimage/libappimage.c]: `data.link` is a `char*` this basically checks if it's  `NULL` or not.

Comment (TheAssassin) [reply src/libappimage/libappimage.c]: Then please do this explicitly, i.e. check against `NULL`. I _hate_ these kind of if clauses where you can only guess what's going on.

Comment (azubieta) [reply src/libappimage/libappimage.c]: Fixed

Comment (TheAssassin) [reply src/libappimage/libappimage.c]: Not really. There's many uses of `data.link`. Please fix all.

Comment (azubieta) [reply src/libappimage/libappimage.c]: Sorry. New changes were pushed

**Comment by TheAssassin** [comment]
Thanks!

**Comment by probonopd** [comment]
Thank you.

**Pull request merged**